111.md

```diff
@@ -0,0 +1,183 @@
+[NIP-11](11.md). All are optional and may be omitted.
+
+All numbers and values show in theses examples, are merely _examples_
+and do not constitute a default value or expectation. The purpose
+of this NIP is to allow clients to directly discover the actual
+values chosen by the relay operators.
+
+Background
+----------
+
+Support for this NIP should be documented in the `supported_nips` list
+of the same response.
+
+```json
+{
+...
+  supported_nips: <a list of NIP numbers supported by the relay, including this one>,
+...
+}
+```
+
+Server Limitations
+------------------
+
+These are limitations imposed by the relay on clients. Your client
+should expect that requests which exceed these *practical* limitations
+are rejected or fail immediately.
+
+```json
+{
+...
+  limitation: {
+        max_raw_json: 16384,
+        max_subscriptions: 20,
+        max_filters: 100,
+        max_limit: 5000,
+        subid_length: 100,
+        min_prefix: 4,
+  }
+...
+}
+```
+
+- `max_raw_json`: this is the maximum number of bytes for incoming JSON that the relay
+will attempt to decode and act upon. When you send large subscriptions, you will be
+limited by this value. It also effectively limits the maximum size of any event. Value is
+calculated from `[` to `]` and is after UTF-8 serialization (so some unicode characters
+will cost 2-3 bytes). It is equal to the maximum size of the WebSocket message frame.
+
+- `max_subscriptions`: total number of subscriptions that may be
+active on a single websocket connection to this relay. It's possible
+that authenticated clients with a (paid) relationship to the relay
+may have higher limits.
+
+- `max_filters`: maximum number of filter values in each subscription.
+Must be one or higher.
+
+- `subid_length`: maximum length of subscription id as a string.
+
+- `min_prefix`: for `authors` and `ids` filters which are to match against
+a hex prefix, you must provide at least this many hex digits in the prefix.
+
+- `max_limit`: the relay server will clamp each filter's `limit` value to this number.
+This means the client won't be able to get more than this number
+of events from a single subscription filter. This clamping is typically done silently
+by the relay, but with this number, you can know that there are additional results
+if you narrowed your filter's time range or other parameters.
+
+
+Event Retention
+---------------
+
+There may be a cost associated with storing data forever, so relays 
+may wish to state retention times. The values stated here are defaults
+for unauthenticated users and visitors. Paid users would likely have
+other policies.
+
+Retention times are given in seconds, with `null` indicating infinity.
+If zero is provided, this means the event will not be stored at
+all, and preferably an error will be provided when those are received.
+
+```json
+{
+...
+  retention: {
+    default: 3600,
+    by_kind: { 0: null, 2: null, 3: null, 4: 172800 },
+  }
+...
+}
+```
```

Comment:
```suggestion
```json
{
...
  retention: [
    { kinds: [0, 1, [5,7], [40,49]], time: 1000 },
    { kinds: [[40000, 49999], time: 100 },
    { kinds: [[30000, 39999], count: 1000 },
    { time: 3600, count: 10000 }
  ]
...
}
```
```

I think this way it is more flexible and one can specify kind ranges, time and count.

Reply:
yes. this is superior. will apply

Reply:
I missed a few ] closing brackets there oops.